test/e2e/apps/job.go

```diff
@@ -510,17 +511,17 @@ done`}
 		framework.ExpectNoError(err, "failed to get latest job object")
 		gomega.Expect(job.Status.Active).Should(gomega.Equal(int32(0)))
 		gomega.Expect(job.Status.Ready).Should(gomega.Equal(ptr.To[int32](0)))
-		gomega.Expect(job.Status.Terminating).Should(gomega.Equal(ptr.To[int32](0)))
```

Comment:
Not a strong view, but I would rather comment it out and add a note:
// TODO(#3998): Restore the assert on status.terminating when PodReplacementPolicy goes GA

I think asserting on fields introduced in another KEP is ok, especially when the features are inter-related, as in this case, because a Job which meets the success criteria still needs to track its terminating Pods. So, I think reverting the assert once JobPodReplacementPolicy goes GA is justified and we can reflect that in a comment.

Reply:
That sounds reasonable. I will do

1. Add TODO comment to each step
2. Add this to the GA graduation criteria to PodReplacementPolicy KEP

Thank you for the suggestion.

Reply:
lgtm, tagged

Reply:
Thank you!